processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -114,3 +129,44 @@ func (e eksDetectorUtils) getConfigMap(ctx context.Context, namespace string, na
 	}
 	return cm.Data, nil
 }
+
+func (e eksDetectorUtils) getClusterName() (string, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return "", err
+	}
+
+	ec2IMS := imds.NewFromConfig(cfg)
+	regionOutput, err := ec2IMS.GetRegion(context.TODO(), &imds.GetRegionInput{})
+	if err != nil {
+		return "", err
+	}
+	cfg.Region = regionOutput.Region
```

Comment:
```suggestion
	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithEC2IMDSRegion())
	if err != nil {
		return "", err
	}
```

Looking at the [default loader in the sdk](https://github.com/aws/aws-sdk-go-v2/blob/v1.21.0/config/config.go#L51), passing in this helper fn should make it query the region from imds.

Reply:
Tested and it works, I'll update. Thanks!